public/app/themes/clarity/src/components/c-article/view.php

```diff
@@ -44,9 +47,12 @@
     <aside class="l-secondary">
     <?php
         echo '<h2 class="o-title">Recent blog posts</h2>';
-        foreach ($query->posts as $key => $post) {
+        $main_post = $post;
```

Comment:
@EarthlingDavey is $post not undefined initially here? I guess it's the Wordpress global $post but the reassignments here are a little odd - is there a cleaner way to do this?

Maybe something like:

```
$recent_query = new WP_Query($query_args->get());
...
if( $recent_query->have_posts()):
    while( $recent_query->have_posts()): $recent_query->the_post();
        include locate_template('src/components/c-article-item/view-blog-feed.php');
    endwhile;
endif;
wp_reset_postdata();
```

Reply:
Thanks for picking this up. Yes, basically you're 100% right :) Your example is the usual WordPress way of doing things.

It's good because it doesn't affect the global $post variable. It's even better if you use `get_template_part` instead of `locate_template` because the variables inside the template file are scoped. 

- We can control what goes in, with the 3rd arg of `get_template_part`
- And whatever variable are assigned in the template don't leak out.

Unfortunately `locate_template` is used throughout the theme, and I'm 99% sure this it quite a poor way to do things. I would like to hear some reasoning for it's use but I expect the dev who chose to do it that way is long gone.

---

I'll spend some time on trying to improve this, it will mean changing 2 loops in additional files, but that's not too excessive :)